### DISCUSSION TITLE
The issue I encountered while deploying a demo via Helm according to the official documentation

### DISCUSSION THREAD

**Discussion by caozhangni** [open-telemetry/opentelemetry-demo#1444] [General]
I encountered an issue while deploying a demo via Helm according to the official documentation(https://opentelemetry.io/docs/demo/kubernetes-deployment/#configure-browser-telemetry). The documentation suggests modifying the **PUBLIC_OTEL_EXPORTER_OTLP_TRACES_ENDPOINT** via **env**, but I found that this causes some environment variables to be lost, leading to abnormal functioning of the frontend component. The list of environment variables in the container is as follows:

> Environment:
>       OTEL_SERVICE_NAME:                                   (v1:metadata.labels['app.kubernetes.io/component'])
>       OTEL_COLLECTOR_NAME:                                my-otel-demo-otelcol
>       OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE:  cumulative
>       PUBLIC_OTEL_EXPORTER_OTLP_TRACES_ENDPOINT:          http://otel.czn.com/otlp-http/v1/traces
>       OTEL_RESOURCE_ATTRIBUTES:                           service.name=$(OTEL_SERVICE_NAME),service.namespace=opentelemetry-demo

After replacing it with **envOverride**, the issue was resolved, and everything worked well:

> Environment:
>       OTEL_SERVICE_NAME:                                   (v1:metadata.labels['app.kubernetes.io/component'])
>       OTEL_COLLECTOR_NAME:                                my-otel-demo-otelcol
>       OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE:  cumulative
>       FRONTEND_PORT:                                      8080
>       FRONTEND_ADDR:                                      :8080
>       AD_SERVICE_ADDR:                                    my-otel-demo-adservice:8080
>       CART_SERVICE_ADDR:                                  my-otel-demo-cartservice:8080
>       CHECKOUT_SERVICE_ADDR:                              my-otel-demo-checkoutservice:8080
>       CURRENCY_SERVICE_ADDR:                              my-otel-demo-currencyservice:8080
>       PRODUCT_CATALOG_SERVICE_ADDR:                       my-otel-demo-productcatalogservice:8080
>       RECOMMENDATION_SERVICE_ADDR:                        my-otel-demo-recommendationservice:8080
>       SHIPPING_SERVICE_ADDR:                              my-otel-demo-shippingservice:8080
>       OTEL_COLLECTOR_HOST:                                $(OTEL_COLLECTOR_NAME)
>       OTEL_EXPORTER_OTLP_ENDPOINT:                        http://$(OTEL_COLLECTOR_NAME):4317
>       WEB_OTEL_SERVICE_NAME:                              frontend-web
>       PUBLIC_OTEL_EXPORTER_OTLP_TRACES_ENDPOINT:          http://otel.czn.com/otlp-http/v1/traces
>       OTEL_RESOURCE_ATTRIBUTES:                           service.name=$(OTEL_SERVICE_NAME),service.namespace=opentelemetry-demo

**Comment by puckpuck** [comment]
This is a docs bug. I made issue #1467 to track getting this fixed.

**Reply by caozhangni** [reply]
Thanks!
